rules/framework/vfs_overlay.bzl

```diff
@@ -30,30 +30,83 @@ def _get_external_contents(prefix, path_str):
 def _get_vfs_parent(ctx):
     return (ctx.bin_dir.path + "/" + ctx.build_file_path)
 
+# Search a VFS json's contents for a name
+def _vfs_json_lookup_contents(contents, name):
+    for c in contents:
+        if name == c["name"]:
+            return c
+    return None
+
+# Builds out the VFS subtrees for a given set of paths. This is useful to
+# construct an in-memory rep of a tree on the file system
+def _build_subtrees(paths, vfs_prefix):
+
+    # This saves having to loop the entire VFS again to translate to
+    # JSON at the trade-off of having extra lookups rarely
+    subdirs = {"contents": [], "type": "directory", "name": "root"}
+    for path_info in paths:
+        path = path_info.framework_path
+
+        parts = path.split("/")
+
+        # current pointer to the current subdirs while walking the path
+        curr_subdirs = subdirs
+
+        # Loop the _framework_ path and add each dir to the current tree.
+        # Assume the last bit is a file then add it as a file  This loop is
+        # quite short ( e.g. O(1) since we loop the VFS path ( e.g. X.h ) and
+        # not the file path.
+        idx = 0
+        for part in parts:
```

Comment:
We may want to take a pass on the "Big O" of this code. See comment for when it len(parts) > 1. Figuring out the issue and getting right VFS produced in the code was the majority of the work here and we may do a pass on the entire VFS construction algorithm runtime now that it's correct.

Reply:
This is now fixed on latest.